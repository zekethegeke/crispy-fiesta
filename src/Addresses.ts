import { WekePageRef } from 'weke-plex';

export interface Address {
    id: number,
    first: string,
    last: string,
    email: string,
    gender: string,
    ip: string
}

export class Addresses {
    readonly males: Address[];
    readonly females: Address[];

    constructor(readonly all: Address[]) {
        this.males = all.filter(it => it.gender === "Male");
        this.females = all.filter(it => it.gender === "Female");
    }

    static ofJson(json: string) {
        return new Addresses(JSON.parse(json) as Address[]);
    }
}

export const addressesPage = new WekePageRef("AddressesPage");
