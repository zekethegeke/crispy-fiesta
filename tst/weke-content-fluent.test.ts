import dedent from 'ts-dedent';
import { Addresses } from '../src/Addresses';
import { wrap } from 'jest-snapshot-serializer-raw';
const addresses = (new Addresses(require("./test-addresses.json")));

class WekeUmlBuilder {
    private content: string;

    constructor(readonly builder: WekeBuilder) {
        this.content = "{{plantUML}}\n@startuml\n";
    }
    append(text: string):  WekeUmlBuilder {
        this.content += text;
        return this;
    }
    endUml() {
        this.content += "\n@enduml\n{{/plantUML}}";
        this.builder.append(this.content)
        return this.builder;
    }

}

class WekeBuilder {
    private content: string;
    constructor() {
        this.content = "";
    }

    h3(title: string): WekeBuilder {
        this.content += `=== ${title} ===\n`;
        return this;
    }
    append(text: string): WekeBuilder {
        this.content += text;
        return this;
    }
    startUml(): WekeUmlBuilder {
        return new WekeUmlBuilder(this);
    }
    build(): string { 
        return this.content;
    }
}

test('smooth talker', () => {
    const males = addresses.males.map(it => `"${it.first}" -> "Male"`);

    const builder = new WekeBuilder();
    const content: string = builder
        .h3("Purpose")
        .append("Our purpose is to test the content.\n")
        .startUml()
            .append("(*) - (*)\n")
            .append(males.join("\n"))
        .endUml()
        .build();

        expect(wrap(content)).toMatchSnapshot();
});