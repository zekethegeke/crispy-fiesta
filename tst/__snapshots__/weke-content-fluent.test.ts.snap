// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smooth talker 1`] = `
=== Purpose ===
Our purpose is to test the content.
{{plantUML}}
@startuml
(*) - (*)
"Tannie" -> "Male"
"Lauren" -> "Male"
"Ervin" -> "Male"
"Hersch" -> "Male"
"Darnall" -> "Male"
"Moise" -> "Male"
"Dunstan" -> "Male"
"Klemens" -> "Male"
"Lancelot" -> "Male"
"Arlan" -> "Male"
"Clemmy" -> "Male"
"Tyrus" -> "Male"
"Colas" -> "Male"
"Dun" -> "Male"
"Demetri" -> "Male"
"Arron" -> "Male"
"Purcell" -> "Male"
"Nigel" -> "Male"
"Pryce" -> "Male"
"Gav" -> "Male"
"Piotr" -> "Male"
"Rickey" -> "Male"
"Wolf" -> "Male"
"Derron" -> "Male"
"Artemas" -> "Male"
"Ernestus" -> "Male"
"Bordy" -> "Male"
"Anatollo" -> "Male"
"Thornie" -> "Male"
"Thorin" -> "Male"
"Neddy" -> "Male"
"Germaine" -> "Male"
"Pablo" -> "Male"
"Aldrich" -> "Male"
"Dale" -> "Male"
"Nappie" -> "Male"
"Nico" -> "Male"
"Shaun" -> "Male"
"Josias" -> "Male"
"Rodolfo" -> "Male"
"Duke" -> "Male"
"Keary" -> "Male"
"Bond" -> "Male"
"Colan" -> "Male"
"Alwin" -> "Male"
"Jackie" -> "Male"
"Christian" -> "Male"
@enduml
{{/plantUML}}
`;
